api/v1/intrusiondetection_types.go

```diff
@@ -91,6 +96,158 @@ type IntrusionDetectionComponentResource struct {
 	ResourceRequirements *corev1.ResourceRequirements `json:"resourceRequirements"`
 }
 
+// IntrusionDetectionControllerDeployment is the configuration for the IntrusionDetectionController Deployment.
+type IntrusionDetectionControllerDeployment struct {
+
+	// Spec is the specification of the IntrusionDetection Deployment.
```

Comment:
Nit: 
after changing the name `IntrusionDetectionDeployment` to ` IntrusionDetectionControllerDeployment`, there are still a number of unchanged comments that refer to `IntrusionDetection Deployment`